### DISCUSSION TITLE
v7.4.0-preview.3 Release of PowerShell

### DISCUSSION THREAD

**Discussion by daxian-dbw** [PowerShell/PowerShell#19549] [Show and tell]
### Breaking Changes

- Remove code related to `#requires -pssnapin` (#19320)

### Engine Updates and Fixes

- Change the arrow used in feedback suggestion to a more common Unicode character (#19534)
- Support trigger registration in feedback provider (#19525)
- Update the `ICommandPredictor` interface to reduce boilerplate code from predictor implementation (#19414)
- Fix a crash in the type inference code (#19400) (Thanks @MartinGC94!)

### Performance

- Speed up `Resolve-Path` relative path resolution (#19171) (Thanks @MartinGC94!)

### General Cmdlet Updates and Fixes

- Infer external application output as strings (#19193) (Thanks @MartinGC94!)
- Fix a race condition in `Add-Type` (#19471)
- Detect insecure `https-to-http` redirect only if both URIs are absolute (#19468) (Thanks @CarloToso!)
- Support `Ctrl+c` when connection hangs while reading data in WebCmdlets (#19330) (Thanks @stevenebutler!)
- Enable type conversion of `AutomationNull` to `$null` for assignment (#19415)
- Add the parameter `-Environment` to `Start-Process` (#19374)
- Add the parameter `-RelativeBasePath` to `Resolve-Path` (#19358) (Thanks @MartinGC94!)
- Exclude redundant parameter aliases from completion results (#19382) (Thanks @MartinGC94!)
- Allow using a folder path in WebCmdlets' `-OutFile` parameter (#19007) (Thanks @CarloToso!)

### Code Cleanup

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@eltociear, @CarloToso</p>

</summary>

<ul>
<li>Fix typo in <code>typeDataXmlLoader.cs</code> (#19319) (Thanks @eltociear!)</li>
<li>Fix typo in <code>Compiler.cs</code> (#19491) (Thanks @eltociear!)</li>
<li>Inline the <code>GetResponseObject</code> method (#19380) (Thanks @CarloToso!)</li>
<li>Simplify <code>ContentHelper</code> methods (#19367) (Thanks @CarloToso!)</li>
<li>Initialize regex lazily in <code>BasicHtmlWebResponseObject</code> (#19361) (Thanks @CarloToso!)</li>
<li>Fix codefactor issue in if-statement (part 5) (#19286) (Thanks @CarloToso!)</li>
<li>Add nullable annotations in <code>WebRequestSession.cs</code> (#19291) (Thanks @CarloToso!)</li>
</ul>

</details>

### Tests

- Harden the default command test (#19416)
- Skip VT100 tests on Windows Server 2012R2 as console does not support it (#19413)
- Improve package management acceptance tests by not going to the gallery (#19412)

### Build and Packaging Improvements

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@dkattan</p>

</summary>

<ul>
<li>Fixing MSI checkbox (#19325)</li>
<li>Update the experimental feature JSON files (#19297)</li>
<li>Update the cgmanifest (#19459, #19465)</li>
<li>Update .NET SDK version to <code>8.0.100-preview.3.23178.7</code> (#19381)</li>
<li>Force updating the transitive dependency on <code>Microsoft.CSharp</code> (#19514)</li>
<li>Update <code>DotnetRuntimeMetadata.json</code> to consume the .NET 8.0.0-preview.3 release (#19529)</li>
<li>Move PSGallery sync to a pool (#19523)</li>
<li>Fix the regex used for package name check in vPack build (#19511)</li>
<li>Make the vPack PAT library more obvious (#19505)</li>
<li>Change <code>Microsoft.CodeAnalysis.CSharp</code> back to 4.5.0 (#19464) (Thanks @dkattan!)</li>
<li>Update to the latest NOTICES file (#19332)</li>
<li>Add <code>PoolNames</code> variable group to compliance pipeline (#19408)</li>
<li>Fix stage dependencies and typo in release build (#19353)</li>
<li>Fix issues in release build and release pipeline (#19338)</li>
</ul>

</details>

[7.4.0-preview.3]: https://github.com/PowerShell/PowerShell/compare/v7.4.0-preview.2...v7.4.0-preview.3

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - E98478BA1A25B10B79BD6B80703865BB802268793CFA2215E53A64A1707E605E
- powershell-7.4.0-preview.3-linux-alpine-x64.tar.gz
  - 157F60F3C687E4C0E9F7A223A2798C4F48AA3523BFFC1AFE76B0F811DC3B790B
- powershell-7.4.0-preview.3-linux-arm32.tar.gz
  - 3B67EA95C2E432A4B12002988DCB72285DBC8801918491F1DBD976B63E599094
- powershell-7.4.0-preview.3-linux-arm64.tar.gz
  - BDDCB22F1A6F1E4262633AD67CF5468EC168BD30A0B5FF64915D24FAEE7D6DB3
- powershell-7.4.0-preview.3-linux-x64.tar.gz
  - CE2FD80BCB50B13D3FEB99AB8E8C69679D80F865B381CE7E57B471275F177748
- powershell-7.4.0-preview.3-linux-x64-fxdependent.tar.gz
  - F6442F77BB1A6A4AE6EDAD98C57E289790399C6E4E16E8430D0FA98F999D2688
- powershell-7.4.0-preview.3-osx-arm64.pkg
  - 2862092490A60C0621646CCD736E46F1FE2304121625174D4894D0FD489B19DE
- powershell-7.4.0-preview.3-osx-arm64.tar.gz
  - B5DDF90BBFA8364A340EDC81D7A60DEDE7B82290CF662C78F8A36C820E1E4D62
- powershell-7.4.0-preview.3-osx-x64.pkg
  - 6F256C74CED7F58540E083FB532D39FA1363F3DC8E870901F34071D5CB88A8DF
- powershell-7.4.0-preview.3-osx-x64.tar.gz
  - EC93CE26D46C546C4665658F11A0F6ED4262785FEEAF7759102584361957A161
- PowerShell-7.4.0-preview.3-win-arm32.zip
  - 920C11F1CC526093127059368111A9D94A1CF80C5FED1417A56629A784D9D39F
- PowerShell-7.4.0-preview.3-win-arm64.zip
  - 9337AC0B9BDDD33EB4BFFF8BF085DF9780A790326CD71CF7C41089A041E11719
- PowerShell-7.4.0-preview.3-win-fxdependent.zip
  - 653E389BFB5512B39D8C5E0540628E5AB8EC7A6B992B08EC044FC32493249659
- PowerShell-7.4.0-preview.3-win-fxdependentWinDesktop.zip
  - CA88C180CD5C56B74D4F2A9D075B8EBC2EE614B99737ADA5D60140DD3EBC2150
- PowerShell-7.4.0-preview.3-win-x64.msi
  - A6E2C02F94135AB9717E49525BBBD359B480ABE1F09AA3505D64E04960F8F672
- PowerShell-7.4.0-preview.3-win-x64.zip
  - 2709BDDE16B769BF10FB67229348AF95E5980A2B6D280BFD99BABAC0B9F96971
- PowerShell-7.4.0-preview.3-win-x86.msi
  - DBC4CA1858F601E4AC68EDD54802BD605CE189B995F23BF74DC7ACEDC496238A
- PowerShell-7.4.0-preview.3-win-x86.zip
  - 93F8A23C489C1E1CF51FD3B6C0BC98A73E8ADAB5148D352881ABBC4C8AC87343
- powershell-preview_7.4.0-preview.3-1.deb_amd64.deb
  - BE418A526D5FE68F39AA1AB126C6262DBF870AE8501BA44CE8B020A2282C536B
- powershell-preview-7.4.0_preview.3-1.cm.x86_64.rpm
  - DED942A32AE0F158BAB47D4188F808627574469197B29A62998F7D7023C572DF
- powershell-preview-7.4.0_preview.3-1.rh.x86_64.rpm
  - A2C9B1242027B4860932CC74D529CBC1FCCEC8235EB865C02E2E36A5737DEE6E

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.4.0-preview.3'>v7.4.0-preview.3 Release of PowerShell</a>.</em>

**Comment by Vintage10E** [comment]
Hi there,

I wanna report an issue with PowerShell v7.4.0-preview.3. The problem comes when you run the `chkdsk /b or chkdsk c: /b` to repair the local disk, in Windows 10 64-bit (latest version). Exactly, the message is as follows:

PowerShell 7.4.0-preview.3
PS C:\Windows\System32> chkdsk c: /b
The type of the file system is NTFS.
Cannot lock current drive.

Chkdsk cannot run because the volume is in use by another
process.  Would you like to schedule this volume to be
checked the next time the system restarts? (Y/N) y

This volume will be checked the next time the system restarts.
**←[31;1mNativeCommandExitException: ←[31;1mProgram "chkdsk.exe" ended with non-zero exit code: 3.←[0m**
PS C:\Windows\System32>

In fact, it does the command that we've applied, but is weird the message. The PowerShell 7.3.4 doesn't give us this issue, so:

PowerShell 7.3.4
PS C:\Users\"User"> chkdsk c: /b
The type of the file system is NTFS.
Cannot lock current drive.

Chkdsk cannot run because the volume is in use by another
process.  Would you like to schedule this volume to be
checked the next time the system restarts? (Y/N) y

This volume will be checked the next time the system restarts.
PS C:\Users\"User">

That's all for now. Thanks.
Sincerely,

@Vintage10E

**Comment by daxian-dbw** [comment]
@Vintage10E It's expected behavior from the `PSNativeCommandErrorActionPreference` experimental feature. See https://learn.microsoft.com/en-us/powershell/scripting/learn/experimental-features?view=powershell-7.3#psnativecommanderroractionpreference for details.

All experimental features are enabled by default for preview packages, and that's why you see the error message.

**Reply by MartinGC94** [reply]
But surely it's not intentional that we see the literal escape sequences rather than the colors they represent?

**Reply by daxian-dbw** [reply]
Right, that part is weird. If the terminal/console in use doesn't support VT, then I presume the error formatting should have known that and not using the VT sequences. It happens on the latest Win 10, so the terminal/console should support VT and PowerShell should have enabled VT at startup. So, the only possibility here seems to be `chkdsk` somehow disabled the VT support in console mode.

**Reply by MartinGC94** [reply]
For me it seems to happen with every executable I test (diskpart, netsh, ping).

**Reply by daxian-dbw** [reply]
Yes, I can reproduce that in conhost, but not in Windows Terminal.
@MartinGC94 can you please help create an issue for this?

**Reply by Vintage10E** [reply]
I've tried out with the commands above, `diskpart, netsh and ping`. **I'm not having issues** applying them on v7.4.0-preview.3.

**Reply by MartinGC94** [reply]
@daxian-dbw Seems like there already are some issues about this: https://github.com/PowerShell/PowerShell/issues/19813 and https://github.com/PowerShell/PowerShell/issues/19425
